src/main/java/org/openstreetmap/atlas/geography/atlas/items/complex/aoi/ComplexAOI.java

```diff
@@ -0,0 +1,185 @@
+package org.openstreetmap.atlas.geography.atlas.items.complex.aoi;
+
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.MultiPolygon;
+import org.openstreetmap.atlas.geography.atlas.items.Area;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasEntity;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+import org.openstreetmap.atlas.geography.atlas.items.complex.ComplexEntity;
+import org.openstreetmap.atlas.geography.atlas.items.complex.RelationOrAreaToMultiPolygonConverter;
+import org.openstreetmap.atlas.tags.filters.TaggableFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+
+/**
+ * Complex Entity for AOI Relations and AOI Areas. AOI relations are those that are
+ * {@link MultiPolygon} and have AOI tags in it and AOI Areas are those {@link Area}s that have AOI
+ * tags in it. AOI tags are checked against the {@link TaggableFilter} passed as an argument or to
+ * the default {@link TaggableFilter} of AOI tags if the tags are not explicitly specified.
+ *
+ * @author sayas01
+ */
+public final class ComplexAOI extends ComplexEntity
+{
+    private static final Logger logger = LoggerFactory.getLogger(ComplexAOI.class);
+    private static final RelationOrAreaToMultiPolygonConverter RELATION_OR_AREA_TO_MULTI_POLYGON_CONVERTER = new RelationOrAreaToMultiPolygonConverter();
+    private static final String AOI_RESOURCE = "aoi-tag-filter.json";
+    private MultiPolygon multiPolygon;
+    // The default AreasOfInterest(AOI) tags
+    private static List<TaggableFilter> defaultTaggableFilter;
+
+    /**
+     * This method creates a {@link ComplexAOI} for the specified {@link AtlasEntity} if it meets
+     * the requirements for Complex AOI relation. The AOI tags are checked against the default tags.
+     *
+     * @param source
+     *            The {@link AtlasEntity} for which the ComplexEntity is created
+     * @return {@link ComplexAOI} if created, else return empty.
+     */
+    public static Optional<ComplexAOI> getComplexAOI(final AtlasEntity source)
```

Comment:
Can this method call `getComplexAOI(source, allTrueFilter)` with a filter that returns true? That might remove some code duplication.

Reply:
I have updated this to call `getComplexAOI(source,  customAoiFilter -> false)` which would remove the duplicated code from this method. Please let me know if this is what you meant by your comment.

Reply:
That's perfect!